### ISSUE TITLE
Refactor and rename of vDVS code

### ISSUE THREAD

**Issue by luomiao** [vmware/vsphere-storage-for-docker#1551]
In order to add the new shared plugin, the current layout will need some changes in order to better utilize common code and eliminate confusions.

A brief plan of the new layout:
```
docker-volume-vsphere
    - (Not mentioned directories will stay the same as before)
    - plugin -> managed_plugin (RENAME)
            this folder hold the Makefile and dockerfile template for building the final docker plugins.
    - vmdk_plugin -> client_plugin (RENAME)
            - vmdk_plugin (NEW)
                   hold existing plugin’s entry main function
            - shared_plugin (NEW)
                   hold the new shared plugin’s entry main function
            - drivers
                    - vmdk
                    - photon
                    - shared (NEW)
            - utils
                    - config
                    - fs
                    - plugin_utils
                    - refcount
                    - log_formatter (NEW)
                    - plugin_server (NEW)
 
```

**Comment by pshahzeb** [comment]
>            - vmdk_plugin (**NEW**) _hold existing plugin’s entry main function_

Since we are renaming this, how about calling it vsphere instead of vmdk

**Comment by luomiao** [comment]
@msterin Your call please! vmdk or vsphere plugin?

**Comment by govint** [comment]
managed_plugin could be plugin_config (thats all it has)

client_plugin could be managed_plugin

vmdk_plugin could be vmdk (its a plugin as we know :-))

shared_plugin could be shared (needs a different name as it looks like its a shared component with the other drivers) or _network_

The names for drivers are a mix of volume type and platform type - vmdk and shared are volume types and photon is a platform which gives vmdks. Could rename so they are all platform perhaps.

**Comment by luomiao** [comment]
To make progress, let's use shared_plugin for now and we can update it when we have a better name!

**Comment by luomiao** [comment]
managed_plugin could be plugin_config: I am ok with this one.
client_plugin could be managed_plugin: so since we have esx_service, this client_plugin means it's the codes that are running on docker clients. So I still think client_plugin is a better name.

**Comment by msterin** [comment]
Client plugin is a generic code, has nothing to do with managed. Current ./plugin is managed plugin and thus is in rename list. Please.keep client_plugin

**Comment by shuklanirdesh82** [comment]
addressed through #1555
